### PR TITLE
fix https://issues.civicrm.org/jira/browse/CRM-19149

### PR DESCRIPTION
**Pull request by wahyukodar**
Fix Financial Transactions Created When Editing a Negative Contribution Incorrect. [https://issues.civicrm.org/jira/browse/CRM-19149](url) .

### CODE REVIEW

**Comment by totten** [comment]
jenkins, ok to test

**Comment by seamuslee001** [comment]
@wahyukodar Wahyu, this fix will break the system for contributions > 0. I think a proper fix is like this https://github.com/civicrm/civicrm-core/compare/master...seamuslee001:CRM-19149?expand=1

File: tests/phpunit/api/v3/ContributionTest.php [hunk 1]
```diff
@@ -1059,6 +1059,31 @@ public function testCreateUpdateContributionPaymentInstrument() {
...
+    $contribution = $this->callAPISuccess('contribution', 'create', $contribParams);
+
+    $newParams = array_merge($contribParams, array(
+        'id' => $contribution['id'],
+        'payment_instrument_id' => $instrumentId,
+      )
+    );
+    $contribution = $this->callAPISuccess('contribution', 'create', $newParams);
+    $this->assertAPISuccess($contribution);
+    $this->_checkFinancialTrxn($contribution, 'paymentInstrument', $instrumentId);
```
Comment (seamuslee001) [inline tests/phpunit/api/v3/ContributionTest.php]: @wahyukodar Wahyu you will need to add an array with total_amount as the key and -100.00 as the value onto the end of this function call, to fix the unit test

**Comment by seamuslee001** [comment]
@wahyukodar whayu once you have fixed up the unit test as per my comment are you able to do git rebase -i origin/master and squish the fix up unit test commits.

**Pull request closed without merging**